code/controllers/subsystems/ticker.dm

```diff
@@ -387,6 +387,11 @@ var/datum/controller/subsystem/ticker/SSticker
 	SSjobs.DivideOccupations() // Apparently important for new antagonist system to register specific job antags properly.
 
 	if(!src.mode.can_start())
+		if(master_mode in list(ROUNDTYPE_STR_RANDOM, ROUNDTYPE_STR_SECRET, ROUNDTYPE_STR_MIXED_SECRET))
+			if(setup_attempts <= 10)
+				mode.fail_setup()
+				mode = null
+				return setup(setup_attempts + 1) // If setup failed, let's try again
```

Comment:
_Really_ not a fan of recursion like this. Maybe make `pregame_tick` handle `setup()` differently? For example:
```dm
	if (pregame_timeleft <= 0 || current_state == GAME_STATE_SETTING_UP)
		current_state = GAME_STATE_SETTING_UP
		wait = 2 SECONDS
		var/setup_res = setup()

		if (setup_res == SETUP_REVOTE)
			wait = 1 SECOND
			is_revote = TRUE
			pregame()
		else if (setup_res == SETUP_REATTEMPT)
			pregame_timeleft = 1 SECOND
			to_world("Reattempting gamemode selection.")
		// else just exits this function and everything goes fine.
```

Reasons, IMO, why this is better:
* Recursion is rarely a good answer.
* You're making the code more resilient to modification by not changing how this function works.